addons/hr_expense/static/src/js/expense_views.js

```diff
@@ -79,7 +82,38 @@ odoo.define('hr_expense.expenses.tree', function (require) {
         }),
     });
 
+    var ExpensesFormController = FormController.extend({
+        /**
+         * @override
+         * This function copies BasicController._notifyInvalidFields instead of calling it through
+         * this._super. This is done to inject the employee_string inside the do_warn notification container instead
+         * of creating a new notification.
+         */
+
+         _notifyInvalidFields: function(invalidFields) {
+            var record = this.model.get(this.handle, {raw: true});
+            const employee_invisible = this.$el.find('.o_expense_form_employee_id')[0].classList.contains('o_invisible_modifier');
```

Comment:
This is quite overkill. Why not raising a classic UserError ? 

1/ This is standard
2/ This doesn't require to update the module (we're on a stable release here).